okhttp/src/main/java/okhttp3/HttpUrl.java

```diff
@@ -1295,7 +1340,12 @@ public HttpUrl build() {
 
       if (encodedQueryNamesAndValues != null) {
         result.append('?');
-        namesAndValuesToQueryString(result, encodedQueryNamesAndValues);
+        if (displayMode != DisplayMode.RAW) {
```

Comment:
Yes, somewhat.  They seem to do some rewriting, but still maintain the reversibility.  I know this is tricky code, so I'm not in a rush to land this, but I think it's useful functionality.